content/doc/administration/requirements/upgrade-java-guidelines.adoc

```diff
@@ -0,0 +1,17 @@
+---
+layout: documentation
+title: Upgrading Jenkins Java version from 8 to 11
+---
+
+In order to upgrade the JVM used to run Jenkins, more specifically from Java 8 to Java 11, there are a few details and steps to take care.
+
+First, we need to understand that some plugins were not directly compatible with Java 11. 
+So if those plugins are installed in our instance, we need to make sure they are installed with those versions or more recent ones.
+
+== Workflow Support 3.0 
```

Comment:
I believe this explanation must be removed actually. 

Given finally the plugin was able to upgrade to Java 11 compatibility, but staying on Java 8, I think it ought to be only discussed in the `workflow-support` changelog.

As this is something likely many, if not most, users will have already done already, independently from Java 11 bump, this might be more misleading here than anything else I think.

---

I think there's two others things:
* remind that agents **must** be running also on Java 11 if the master is upgraded (point to versioncolumn plugin?)
* a new the detached `jaxb` plugin will be installed automatically on Java 11 runtime.

Reply:
Oh, and backup backup backup :)

Reply:
> As this is something likely many, if not most, users will have already done already

I'd prefer not relying on something un-told. In the end, the upgrade is required anyway and should have been done already.

---

Noted the two other points, I'll add them.

Reply:
As discussed off-the-band, you have a point. What I am saying is that anyway, we should be prescriptive, and basically say like "please first upgrade every single plugins you have before considering a JVM upgrade, especially make sure that `workflow-support` is 3.0+.".

This way: if there's an plugins upgrade issue while still on Java 8, then users will not be inclined to mix it with a JVM upgrade issue. 
This will be easier for them to fix if so, and we'll get less/no misleading reports about failed Java 11 upgrades. Win win.